docs/architecture/ADR-005-QGB_posting_confirms_in_reserved_namespace.md

```diff
@@ -0,0 +1,97 @@
+# ADR 5: Reduce QGB state
+
+## Changelog
+
+- {date}: {changelog}
+
+## Context
+
+Currently, the QGB is using the state to save all the attestations and confirms it needs.
+We propose to post `Valset Confirms` and `DataCommitment Confirms` in a reserved namespace instead of adding them to the state to achieve the following:
+
+- Reduce the state machine complexity
+- Reduce the amount of state used by the QGB
+- Prepare for a more Rollup oriented QGB design
+
+## Alternative Approaches
+
+### Keep the existing design
+
+Keeping the current design would entail using the state extensively.
+This proves bad when the state grows after a few hundred attestations, and performing checks on the `Valset Confirms` and `DataCommitment Confirms` becomes too expensive.
+An example of such issue in here: [QGB data commitments/valsets state machine checks #631](https://github.com/celestiaorg/celestia-app/issues/631) and [Investigate the QGB transactions cost #603](https://github.com/celestiaorg/celestia-app/issues/603).
+
+The approach that we were planning to take is to cleanup the state after the unbonding period.
+This way, we will always have a fixed sized state instead of a gigantic one, issue defining this: [Prune the QGB state after the unbonding period ends #309](https://github.com/celestiaorg/celestia-app/issues/309).
+This would mean that the QGB contracts deployed after genesis will never have the whole history of attestations.
+
+### Separate P2P network
+
+This would mean gossiping the confirms and attestations in a separate P2P network.
+The pros of this approach are that it will be cheaper and wouldn't involve any state changes.
+However, slashing will not be possible.
```

Comment:
> slashing will not be possible.

perhaps "very difficult" is more accurate.

the main issue from my point of view with this approach is that it is a different set of liveness guarantees. We have no way of knowing if the orchestrator signed in time as we have no permanent record of when they signed.

Reply:
```suggestion
However, slashing will be very difficult especially for liveness, i.e. an orchestrator not signing an attestation then slash them after a certain period.
```

Reply:
literally perfect!